### PR TITLE
feat: improved threshold volume API and refactored boundingBox utils

### PR DESCRIPTION
**Pull request by sedghi**
- cleanup volume API to not rely on annotations
- refactor bounding box utilities
- make viewport parameter in pointInShereOptional

### CODE REVIEW

File: packages/tools/src/utilities/pointInSurroundingSphereCallback.ts [hunk 1]
```diff
@@ -4,33 +4,134 @@ import type { Types } from '@cornerstonejs/core';
...
  */
 export default function pointInSurroundingSphereCallback(
-  viewport: Types.IVolumeViewport,
   imageData: vtkImageData,
   circlePoints: [Types.Point3, Types.Point3],
-  callback: PointInShapeCallback
+  callback: PointInShapeCallback,
+  viewport?: Types.IVolumeViewport
 ): void {
+  // We can run the sphere equation to determine if a point is inside
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/pointInSurroundingSphereCallback.ts:32]: Is an estimation good enough? I don't get how working this out in ijk space is useful when these dimensions could be e.g. (1,1,3) in many very common cases, you need to do all these calculations in world space, no?

File: packages/tools/src/utilities/pointInSurroundingSphereCallback.ts [hunk 2]
```diff
@@ -4,33 +4,134 @@ import type { Types } from '@cornerstonejs/core';
...
+
+  if (!viewport) {
+    // If no viewport is provide (no camera), we can estimate the bounding box
+    // of the sphere in index space.
+    // This is done by calculating the maximum value for radius in the index
+    // space (since the radius is in world space, we need to convert it to index, and
+    // each dimensions can have a different scale factor). Therefore, by finding
+    // the minimum spacing value in the imageData, we can calculate the maximum
+    // radius in index space and use that to calculate the bounds of the sphere
+    // This will not be accurate, but it is a good first approximation.
```
Comment (JamesAPetts) [inline packages/tools/src/utilities/pointInSurroundingSphereCallback.ts:86]: Considering one of the dimensions can be 2-10x bigger than the others, I'm not sure this is a good approximation?

**Review by JamesAPetts — CHANGES_REQUESTED** [review]
I'm not so sure about the approximation used here for point in sphere and think we probably just need to explicitly check the vector distance in smaller than some value? However, perhaps you have some rationalle for this approximation, we can discuss tomorrow.

File: packages/tools/src/utilities/boundingBox/getBoundingBoxAroundShape.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  xMax = Math.floor(xMax);
+  yMin = Math.floor(yMin);
+  yMax = Math.floor(yMax);
+  zMin = Math.floor(zMin);
+  zMax = Math.floor(zMax);
+
+  if (dimensions) {
+    const [width, height, depth] = dimensions;
+    xMin = Math.max(0, xMin);
+    xMax = Math.min(width - 1, xMax);
```
[the remark this answers is not in the document]
Comment (sedghi) [reply packages/tools/src/utilities/boundingBox/getBoundingBoxAroundShape.ts:42]: xMin, xMax, .... are in IJK, we use dimensions to clip

[the remark this answers is not in the document]
Comment (JamesAPetts) [reply packages/tools/src/utilities/boundingBox/getBoundingBoxAroundShape.ts:43]: Oh so this is inclusive, looking at this again, this makes sense.

**Review by JamesAPetts — APPROVED** [review]
LGTM, just some discussion points but good to go.

**Pull request merged**